src/Microsoft.Sbom.Api/Config/ConfigSanitizer.cs

```diff
@@ -84,14 +91,43 @@ private ConfigurationSetting<IList<ManifestInfo>> GetDefaultManifestInfoForValid
         }
 
         return new ConfigurationSetting<IList<ManifestInfo>>
+            {
+                Source = SettingSource.Default,
+                Value = new List<ManifestInfo>()
+                {
+                    defaultManifestInfo
+                }
+            };
+        }
+
+    private void ValidateBuildDropPathConfiguration(IConfiguration configuration)
         {
-            Source = SettingSource.Default,
-            Value = new List<ManifestInfo>()
+            if (configuration.ManifestToolAction != ManifestToolActions.Validate && configuration.ManifestDirPath?.Value != null && configuration.DockerImagesToScan?.Value != null)
             {
-                defaultManifestInfo
+                return;
             }
-        };
-    }
+            else if (configuration.ManifestToolAction != ManifestToolActions.Validate && configuration.ManifestDirPath?.Value == null && configuration.BuildComponentPath?.Value == null && configuration.DockerImagesToScan?.Value != null)
+            {
+                throw new ValidationArgException($"Please provide a (-m) if you intend to create an SBOM with only the contents of the Docker image or a (-bc) if you intend to include other components in your SBOM.");
+            }
+            else if (configuration.ManifestToolAction != ManifestToolActions.Validate && configuration.ManifestDirPath?.Value == null && configuration.DockerImagesToScan?.Value != null)
+            {
+                throw new ValidationArgException($"Please provide a value for the ManifestDirPath (-m) parameter to generate the SBOM for the specified Docker image.");
+            }
+            else
+            {
+                throw new ValidationArgException($"Please provide a value for the BuildDropPath (-b) parameter to generate the SBOM.");
```

Comment:
This exception message might appear during validation with these parameters. These parameters are not particularly valid, but exception message for empty -b param should not confuse customer about validation vs generation.
```
>> Microsoft.Sbom.Tool.exe Validate -b "" -o X:\test\output.json -mi SPDX:2.2 -v Verbose
Please provide a value for the BuildDropPath (-b) parameter to generate the SBOM.
```